connector_flow/task/abstract_task.py

```diff
@@ -0,0 +1,121 @@
+# -*- coding: utf-8 -*-
+##############################################################################
+#
+#    Copyright (C) 2014 initOS GmbH & Co. KG (<http://www.initos.com>).
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU Affero General Public License as
+#    published by the Free Software Foundation, either version 3 of the
+#    License, or (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+from openerp.addons.connector.session import ConnectorSession
+from openerp import _
+import simplejson
+from base64 import b64encode, b64decode
+
+
+class abstract_task(object):
+
+    def __init__(self, cr, uid, ids):
+        self.session = ConnectorSession(cr, uid)
+        self._id = ids[0]
+
+    def run_task(self, task_id, **kwargs):
+        return self.session.env['impexp.task'].browse(task_id).do_run(**kwargs)
+
+    def related_action(self, job=None, **kwargs):
+        """Overwrite this method to add a related action function
+           for a specific task type."""
+        pass
+
+    def run(self, **kwargs):
+        """All the task core action happens here"""
+        raise Exception("Not Implemented")
+
+    def run_successor_tasks(self, **kwargs):
+        successors = self.session.env['impexp.task.transition'].\
+            search_read([('task_from_id', '=', self._id)], ['task_to_id'])
+        retval = None
+        for succ in successors:
+            retval = self.run_task(succ['task_to_id'][0], **kwargs)
+        return retval
+
+    def create_file(self, filename, data):
+        ir_attachment_id = self.session.env['ir.attachment'].\
+            create({'name': filename,
+                    'datas': b64encode(data),
+                    'datas_fname': filename})
+        file_id = self.session.env['impexp.file'].\
```

Comment:
Rename `file_id` to `impexp_file` because it is a record (and `file` is a builtin)